### PR TITLE
cookbook - gaussian process classifier

### PR DESCRIPTION
**Pull request by OXPHOS**

### CODE REVIEW

File: doc/cookbook/source/examples/multiclass_classifier/gaussian_process_classifier.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
+===========================
+Gaussian Process Classifier
```
Comment (karlnapf) [inline doc/cookbook/source/examples/multiclass_classifier/gaussian_process_classifier.rst]: can you put this under the GP folder? Not under multiclass?

File: doc/cookbook/source/examples/multiclass_classifier/gaussian_process_classifier.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+.. sgexample:: gaussian_process_classifier.sg:train_and_apply
+
+We can evaluate test performance via e.g. :sgclass:`CMulticlassAccuracy`.
+
+.. sgexample:: gaussian_process_classifier.sg:evaluate_accuracy
+
+----------
+References
+----------
+:wiki:`Gaussian_process`
```
Comment (karlnapf) [inline doc/cookbook/source/examples/multiclass_classifier/gaussian_process_classifier.rst]: reference to williams and barber, there is a chaper on multiclass laplace approximation in there, the pdf is free online

**Comment by karlnapf** [comment]
integration data

**Comment by OXPHOS** [comment]
@karlnapf updated

File: doc/cookbook/source/examples/gaussian_processes/gaussian_process_classifier.rst [hunk 3]
```diff
@@ -0,0 +1,51 @@
+===========================
+Gaussian Process Classifier
+===========================
+
+Application of Gaussian process in binary and multi-class classification.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/gaussian_processes/gaussian_process_classifier.rst]: Gaussian processes

File: doc/cookbook/source/examples/gaussian_processes/gaussian_process_classifier.rst [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+`See Gaussian process regression cookbook
+<http://shogun.ml/cookbook/latest/examples/gaussian_processes/gaussian_process_regression.html>`_
+and :cite:`Rasmussen2005GPM` for more information on Gaussian process.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/gaussian_processes/gaussian_process_classifier.rst]: processes

File: doc/cookbook/source/examples/gaussian_processes/gaussian_process_classifier.rst [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+-------
+Example
+-------
+
+Imagine we have files with training and test data. We create CDenseFeatures (here 64 bit floats aka RealFeatures) and :sgclass:`CMulticlassLabels` as
+
+.. sgexample:: gaussian_process_classifier.sg:create_features
+
+To fit the input (training) data :math:`\mathbf{X}`, we have to choose appropriate :sgclass:`CMeanFunction`
+and  :sgclass:`CKernel`. Here we use a basic :sgclass:`CCountMean` and a :sgclass:`CGaussianKernel` with chosen width parameter.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/gaussian_processes/gaussian_process_classifier.rst]: Const mean

File: doc/cookbook/source/examples/gaussian_processes/gaussian_process_classifier.rst [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+.. sgexample:: gaussian_process_classifier.sg:create_appropriate_kernel_and_mean_function
+
+We need to specify the inference method to find the posterior distribution of the function values :math:`\mathbf{f}`.
+Here we choose to perform Laplace approximation inference method with an instance of :sgclass:`CMultiLaplaceInferenceMethod` (See Chapter 18.2 in :cite:`barber2012bayesian` for a detailed introduction) 
+and pass it the chosen kernel,
+the training features, the mean function, the labels and an instance of :sgclass:`CSoftMaxLikelihood`,
+to specify the distribution of the targets/labels as above.
+Finally we generate a :sgclass:`CGaussianProcessClassification` class to be trained.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/gaussian_processes/gaussian_process_classifier.rst]: not class, it is an object

File: doc/cookbook/source/examples/gaussian_processes/gaussian_process_classifier.rst [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+.. sgexample:: gaussian_process_classifier.sg:create_instance
+
+Then we can train the model and evaluate the predictive distribution.
+We get predicted :sgclass:`CMulticlassLabels`.
+
+.. sgexample:: gaussian_process_classifier.sg:train_and_apply
+
+We cane extract the probabilities:
```
Comment (karlnapf) [inline doc/cookbook/source/examples/gaussian_processes/gaussian_process_classifier.rst]: cane sugar :)

File: examples/meta/src/gaussian_processes/gaussian_process_classifier.sg [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+gp_classifier.train()
+MulticlassLabels labels_predict = gp_classifier.apply_multiclass(features_test)
+#![train_and_apply]
+
+#![extract_the_probabilities]
+RealVector prob = gp_classifier.get_probabilities(features_train)
+#![extract_the_probabilities]
+
+#![evaluate_accuracy]
+MulticlassAccuracy eval()
```
Comment (karlnapf) [inline examples/meta/src/gaussian_processes/gaussian_process_classifier.sg]: pls  dont call this eval but different

**Comment by karlnapf** [comment]
Only minor changes remaining. Pls update, squash, and we can merge 
@yorkerlin check out the [link](http://buildbot.shogun-toolbox.org/cookbook_pr/6f02ef208dff1d8d90d09358f6a30d2af9d68431/examples/gaussian_processes/gaussian_process_classifier.html) Good how your stuff looks like, no? :)

**Comment by OXPHOS** [comment]
@karlnapf ready.
`eval` is used in all cookbooks. We may have to change them all if there's conflict..

**Comment by karlnapf** [comment]
Yep, we can change that, it is an easy grep....

**Pull request merged**